### PR TITLE
Feature/rework periodic value deserialisation

### LINKED ISSUE #151: Rework periodic value deserialisation
Some changes are necessary. Predecessor was #124.

Some parameters have to have a specific length due to the implementation of multi-period approach, e.g for every timestep or just for every period. Periodically changing values are indicated in datapackage as `list`.

[The reading routine](https://github.com/oemof/oemof-tabular/blob/9f68961dabcbb2d39a972fafebaa4f1b52406ce7/src/oemof/tabular/datapackage/reading.py#L474) is checking whether the number of values in the list are equal to the number of periods. If so, they are converted to a timeseries for all timesteps. In some cases this is not necessary and only one value per period is sufficient. 

The following parameter conversions have to be checked or maybe implemented:
For sake of clarity, creating a dict to define to which format a parameter is converted might be useful. 

sth like this pseudo code
```
conversion_parameters = {"fixed_costs" : create_periodic_values, ...}
if key in conversion_parameters.keys():
  facade_value =  conversion_parameters[key](value)
```
  
Parameters to catch:

- yearly:
  - [x] fixed_costs
  - [x] marginal_costs
 
 - periodically:
   - [x] capacity
   - [x] capacity_cost
   - [x] capacity_potential

### PR DESCRIPTION
**Pull request by FelixMau**
# Description

To implement multiperiod feature multiple values might be passed for one facade. Therefore implemented array values that need to be unpacked to either act as periodical values, yearly values or as full series. 

Fixes #151 

## Type of change

Please tick or delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

Please tick or delete options that are not relevant.

- [x] New and adjusted code is formatted using the `pre-commit` hooks
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] I have added new features/fixes to the [CHANGELOG](https://github.com/oemof/oemof-tabular/tree/dev/CHANGELOG.rst)
- [ ] I have added my name to [AUTHORS]((https://github.com/oemof/oemof-tabular/tree/dev/AUTHORS.rst)
)

### CODE REVIEW

**Comment by FelixMau** [comment]
Unfortunately test failes when a periodical changing value for "capacity" is added.

Deserializing for 
- storage 'capacity_cost'
- link 'capacity_cost'
- dispatchable 'capacity' (breaking and removed in #1be182b)
- volatile 'marginal_cost'

**Comment by FelixMau** [comment]
Setting Capacity to periodical changing value failed as intended by solph when building flows. 
Changing nominal values are only allowed if Investment is set. 

Therefore added capacity periodical values to storage (since it is Investment object) 

---------------
Note: Improvment in [_flow Code style](https://github.com/oemof/oemof-solph/blob/23bceff8a1561b1c9dad1756aca99834524deb11/src/oemof/solph/flows/_flow.py#L187-L199)  using else instead of 'pre setting' values could be improved. Throwing error might also be useful sometime

File: setup.py [hunk 1]
```diff
@@ -70,6 +70,7 @@ def read(*names, **kwargs):
         # "oemof.solph>=0.5.1",
         "oemof.solph==0.5.2dev0",
         "pandas>=0.22",
+        "oemof.network==0.5.0a4",
```
Comment (iuljan) [inline setup.py]: Can you plese add a comment like: "Temporal fix, due to breaking changes in 0.5.1"?

File: src/oemof/tabular/datapackage/reading.py [hunk 2]
```diff
@@ -433,15 +434,108 @@ def create_periodic_values(values, periods_index):
...
-        for i in range(len(years) - 1):
-            diff = years[i + 1] - years[i]
+        for i in range(len(period_years) - 1):
+            diff = period_years[i + 1] - period_years[i]
             period_results = pd.Series(repeat(values[i], diff))
             results = pd.concat([results, period_results])
         results = pd.concat([results, pd.Series(values[-1])])
         return results.tolist()
 
+    def convert_and_deserialize_dtypes(facade, period_data):
```
Comment (iuljan) [inline src/oemof/tabular/datapackage/reading.py]: How about `convert_sequences()` ? I am not sure about the `dtypes` thing here as the type of sequences is mainly chosen by `parameter-name` isnt it?

File: src/oemof/tabular/datapackage/reading.py [hunk 3]
```diff
@@ -433,15 +436,108 @@ def create_periodic_values(values, periods_index):
...
+        facade
+        """
+
+        yearly_values = ["fixed_costs", "marginal_costs"]
+        periodical_values = [
+            "capacity",
+            "capacity_cost",
+            "capacity_potential",
+            "storage_capacity",
+        ]
```
Comment (iuljan) [inline src/oemof/tabular/datapackage/reading.py:491]: this is great!

**Review by iuljan — APPROVED** [review]
This looks great. Thanks a lot

File: src/oemof/tabular/datapackage/reading.py (hunk 2 above)
Comment (FelixMau) [reply src/oemof/tabular/datapackage/reading.py]: Named it "unpack_sequences()"

**Pull request merged**